src/amber/pipes/client_ip.cr

```diff
@@ -0,0 +1,34 @@
+module Amber
+  module Pipe
+    # Handler to determine real client IP address when Amber is behind a
+    # reverse proxy and save it to context.client_ip.
+    #
+    # The proxy should pass client IP in a header such as "X-Forwarded-For".
+    # In case of multiple values in a header, the first one is used.
+    #
+    # Crystal does not currently make remote IP available, so it is impossible
+    # to accept the header value only when coming from trusted proxy IPs.
+    # For that and other benefits, it is suggested to always run Amber apps
+    # behind proxies such as HAProxy or Nginx.
+    class ClientIp < Base
+      def initialize(header : String = "X-Forwarded-For")
+        @headers = [header]
+      end
+      def initialize( @headers : Array(String))
+      end
+
+      def call(context : HTTP::Server::Context)
+        @headers.each do |header|
+          if addresses = context.request.headers.get?(header)
```

Comment:
@amberframework/core-team @amberframework/contributors This is a little out of the scope of this pull request, but HTTP headers are technically [case ignorant](http://httpwg.org/specs/rfc7230.html#header.fields). Should we be doing something somewhere to downcase and internally standardize all http headers? It might help avoid [writing downcase everywhere](https://github.com/rack/rack/search?utf8=%E2%9C%93&q=downcase&type=) and I think coming out of the gate with a strong opinion on this might have some long term benefits.